src/util/nodeSelectors.ts

```diff
@@ -54,6 +54,22 @@ export function simpleSelectionExtractor(
   };
 }
 
+export function selectNamedChild(childIdx: number) {
+  return function (editor: TextEditor, node: SyntaxNode): SelectionWithContext {
+    var child = node.namedChild(childIdx);
+    if (child == null) {
+      child = node;
+    }
```

Comment:
Why do we revert to the parent if we can't find a child at the given index?

Reply:
I didn't really know what to do here because it looks like all the nodeSelectors always return a selection. Can I return `null`?

Reply:
This is the wrong place to do that. The node finder can return a node or null. The purpose of the selector is to extract the correct range from at give node.

Reply:
@AndreasArvidsson I guess my dilemma here is, not every node has a child right? So what selection should I return in that case? Open to any suggestions on how this should be structured

Reply:
No there are leaf nodes without any children. The selector should be called with a correct node from the beginning. Find the correct node before calling the selector. You should probably add this logic to childAtIndexMatcher.

Reply:
Let me know if you have any problem with this and I will give you a hand. but basically you have to check that the node return by patternFinder actually has that child. Then pass the child itself to the selector. If you don't need any specific logic in the selector you can use the simple one which is default in the matcher.

Reply:
@AndreasArvidsson thanks! I think I understand what you are suggesting now

Reply:
@AndreasArvidsson fixed this, thanks for this suggestion!

Reply:
Now it's looking like you are not using the patterns argument? This means that it will match any node at the correct index no mother of the parent type. I would probably solve with something like this:
```ts
export function childAtIndexMatcher(patterns: string[], childIdx: number): NodeMatcher {
  const finder = patternFinder(...patterns);
  return matcher(
    (node: SyntaxNode) => finder(node)?.namedChild(childIdx) ?? null
  );
}
```

Reply:
@AndreasArvidsson used your suggestion